src/NLog/Logger.cs

```diff
@@ -142,6 +166,28 @@ public void SetProperty(string propertyKey, object propertyValue)
             Properties[propertyKey] = propertyValue;
         }
 
+        /// <summary>
+        /// Updates the specified context properties for the current logger. The logger will append it for all log events.
+        ///
+        /// With <see cref="Properties"/> property, all properties can be enumerated (or updated). 
+        /// </summary>
+        /// <remarks>
+        /// It is highly recommended to ONLY use <see cref="WithProperties(IEnumerable{KeyValuePair{string, object}})"/> for modifying context properties.
+        /// This method will affect all locations/contexts that makes use of the same named logger object. And can cause
+        /// unexpected surprises at multiple locations and other thread contexts.
+        /// </remarks>
+        /// <param name="properties">Collection of key-value pair properties</param>
+        public void SetProperties(IEnumerable<KeyValuePair<string, object>> properties)
```

Comment:
Please remove this method.

Reply:
I don't think a inconsistent API is a good idea.

Reply:
Don't worry. `SetProperty` is now made obsolete